packages/amplify-category-auth/resources/cloudformation-templates/auth-template.yml.ejs

```diff
@@ -589,16 +589,27 @@ Resources:
             - '   let providerCredsIndex = hostedUIProviderCreds.findIndex((provider) => provider.ProviderName === providerName);'
             - '   let providerCreds = hostedUIProviderCreds[providerCredsIndex];'
             - '   let requestParams = {'
-            - '    ProviderDetails: {'
-            - '     ''client_id'': providerCreds.client_id,'
-            - '     ''client_secret'': providerCreds.client_secret,'
-            - '     ''authorize_scopes'': providerMeta.authorize_scopes'
-            - '    },'
             - '    ProviderName: providerMeta.ProviderName,'
             - '    UserPoolId: userPoolId,'
-            - '    AttributeMapping: providerMeta.AttributeMapping'
+            - '    AttributeMapping: providerMeta.AttributeMapping,'
             - '   };'
-            - '   return requestParams;'
+            - '   let providerDetails;'
+            - '   if (providerMeta.ProviderName === ''SignInWithApple'') {'
+            - '    providerDetails = {'
```

Comment:
Can't this be `requestParams.ProviderDetails = { ... };`. Then you don't need the `providerDetails` variable or the spread operation on line 612.

Reply:
updated